compiler/frontend/src/org/jetbrains/kotlin/parsing/KotlinParsing.java

```diff
@@ -112,9 +112,20 @@ void parseFile() {
             parseTopLevelDeclaration();
         }
 
+        checkUnclosedBlockComment();
         fileMarker.done(KT_FILE);
     }
 
+    private void checkUnclosedBlockComment() {
+        if (TokenSet.create(BLOCK_COMMENT, DOC_COMMENT).contains(myBuilder.rawLookup(-1))) {
+            int startOffset = myBuilder.rawTokenTypeStart(-1);
+            int endOffset = myBuilder.rawTokenTypeStart(0);
+            if (!myBuilder.getOriginalText().subSequence(startOffset, endOffset).toString().endsWith("*/")) {
+                error("Unclosed comment");
```

Comment:
Looks like this fixes it:

```
                PsiBuilder.Marker marker = myBuilder.mark();
                marker.error("Unclosed comment");
                marker.setCustomEdgeTokenBinders(WhitespacesBinders.GREEDY_RIGHT_BINDER, WhitespacesBinders.GREEDY_RIGHT_BINDER);
```

But I think the Java version uses some other technique